src/features/iterator-helpers.md

```diff
@@ -0,0 +1,188 @@
+
+// Counts posts with security tag.
+const count = tags.reduce((sum , value) => sum + (value === 'security' ? 1 : 0), 0);
+```
+
+## .toArray()
+
+`toArray` returns an array form iterator values. 
+
+```javascript
+// Select the list of blog posts from a blog archive page.
+const posts = document.querySelectorAll('li');
+
+// Create an array from the list of 10 recent blog posts.
+const arr = posts.values().take(10).toArray();
+```
+
+## .forEach(fn)
+
+`forEach` takes a function as an argument and is applied on each element of the iterator. This helper is called for its side effect and returns `undefined`.
+
+```javascript
+// Select the list of blog posts from a blog archive page.
+const posts = document.querySelectorAll('li');
+
+// Get the dates that at least one blog post is published and log them.
+const dates = new Set();
+const forEach = posts.values().forEach((x) => dates.add(x.querySelector('time')));
+console.log(dates);
+```
+
+## .some(fn)
+
+`some` takes a predicate function as an argument. This helper returns `true` if any iterator element returns true when the function is applied to it. The iterator is consumed after `some` is called.
+
+```javascript
+// Select the list of blog posts from a blog archive page.
+const posts = document.querySelectorAll('li');
+
+// Find out if text content (title) of any blog post includes the `Iteartors` keyword.
+posts.values().some((x) => x.textContent.includes('Iterators'));
+```
+
+## .every(fn)
+
+`every` takes a predicate function as an argument. This helper returns `true` if every iterator element returns true when the function is applied to it. The iterator is consumed after `every` is called.
+
+```javascript
+// Select the list of blog posts from a blog archive page.
+const posts = document.querySelectorAll('li');
+
+// Find out if text content (title) of all blog post includes the `V8` keyword.
+posts.values().every((x) => x.textContent.includes('V8'));
+```
+
+## .find(fn)
+
+`find` takes a predicate function as an argument. This helper returns the first value of the iterator for which the function returns a truthy value, or `undefined` if no value of the iterator does.
+
+```javascript
+// Select the list of blog posts from a blog archive page.
+const posts = document.querySelectorAll('li');
+
+// Log the text content (title) of the recent blog post includes `V8` keyword.
+console.log(posts.values().find((x) => x.textContent.includes('V8')).textContent);
+```
+
+## Iterator.from(object)
+
+`from` is a static method and takes an object as an argument. If the `object` is already an instance of Iterator, the helper returns it directly. If the `object` has @@iterator which means it is an iterable, its @@iterator method is called to get the iterator and the helper returns it. Otherwise, a new Iterator object (that inherit from Iterator.prototype and has `next()` and `return()` methods) is created that wraps the `object` and is returned by this helper.
```

Comment:
```suggestion
`from` is a static method and takes an object as an argument. If the `object` is already an instance of Iterator, the helper returns it directly. If the `object` has `Symbol.iterator`, which means it is an iterable, its `Symbol.iterator` method is called to get the iterator and the helper returns it. Otherwise, a new `Iterator` object (that inherit from `Iterator.prototype` and has `next()` and `return()` methods) is created that wraps the `object` and is returned by this helper.
```

The @@ syntax isn't very helpful and the spec editors are trying to move away from it in favor of just typing out `Symbol.foo`.